habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDepsMojo.java

```diff
@@ -138,98 +99,88 @@ public void execute() throws MojoExecutionException, MojoFailureException {
-                Files.createDirectories(this.anchorOutputDirectory.resolve(relativePath).getParent());
-                Files.copy(srcRoot.resolve(relativePath), this.anchorOutputDirectory.resolve(relativePath), StandardCopyOption.REPLACE_EXISTING);
-            }
+        logger.info("Staging {} monorepo dependency files from {}.",
+                fileSetManager.getIncludedFiles(sourceFileSet).length,
+                relativeProjectPath.getFileName()
+        );
+        for (String includedFile : fileSetManager.getIncludedFiles(sourceFileSet)) {
+            Path relativeFilePath = relativeProjectPath.resolve(includedFile);
+            Files.createDirectories(destRoot.resolve(relativeFilePath).getParent());
+            Files.copy(sourceRoot.resolve(relativeFilePath), destRoot.resolve(relativeFilePath));
         }
-        return primaryProjectPath;
     }
 
-    private Path getWorkingDirectoryPath(MavenProject project) {
-        return project.getBasedir().toPath().resolve(workingDirectoryRelativeToBasedir);
-    }
+    protected FileSet getSourceSet() {
+        if( defaultSourceSet != null ) {
+            return defaultSourceSet;
+        }
 
-    private FileSet getDefaultFileSet(MavenProject project) throws MojoExecutionException {
         FileSet fileSet = new FileSet();
-        fileSet.addExclude(distDirectoryRelativeToBasedir + GLOB_RECURSIVE_ALL);
-        fileSet.addExclude(targetDirectoryRelativeToBasedir + GLOB_RECURSIVE_ALL);
-
-        // find the virtual environment path for the given Habushu-packaged project
-        String virtualEnvironmentPath = HabushuUtil.findCurrentVirtualEnvironmentFullPath(
-                pythonVersion,
-                usePyenv,
-                new File(project.getBuild().getDirectory() + patchInstallScriptRelativeToBuildDirectory),
-                new File(project.getBasedir() + workingDirectoryRelativeToBasedir),
-                rewriteLocalPathDepsInArchives,
-                getLog()
-        );
-        virtualEnvironmentPath = HabushuUtil.getCleanVirtualEnvironmentPath(virtualEnvironmentPath);
-
-        // if a valid virtual environment was found, relativize and add to the fileSet for exclusions
-        if (virtualEnvironmentPath != null && !virtualEnvironmentPath.isEmpty()
-                && new File(virtualEnvironmentPath).isDirectory()) {
-            Path venvDirRelativeToBasedir = getWorkingDirectoryPath(project)
-                    .relativize(Paths.get(virtualEnvironmentPath));
-            fileSet.addExclude(venvDirRelativeToBasedir + GLOB_RECURSIVE_ALL);
-        }
+        Path srcPath = sourceDirectory.toPath();
+        Path basePath = project.getBasedir().toPath();
+        Path relativeSrc = basePath.relativize(srcPath);
+        fileSet.addInclude(relativeSrc +"/**");
+        fileSet.addInclude("pyproject.toml");
+        fileSet.addInclude("poetry.toml");
+        fileSet.addInclude("poetry.lock");
+        fileSet.addInclude("README.md");
```

Comment:
I: This is the primary change in this PR.  Instead of discovering all of the excludes, we drive the fileset based on including the configured source directory and the necessary poetry project files.  The readme is unfortunately required to execute `poetry bundle venv`, so we set it to the default README that poetry generates for new projects.